tests/unit/test_application.py

```diff
@@ -398,7 +400,13 @@ def test_esm_error(snapcraft_yaml, base, monkeypatch, capsys):
     application.main()
 
     _, err = capsys.readouterr()
-    assert f"Base {base!r} is not supported by this version of Snapcraft" in err
+
+    assert re.match(
```

Comment:
I modified these tests because they were not guarding against these being internal errors.